File: backend/src/main/java/isep/webapp/terasync/model/user/Business.java
package isep.webapp.terasync.model.user;

import isep.webapp.terasync.model.RatingList;
import isep.webapp.terasync.model.odSystem.CustomerOrder;
import isep.webapp.terasync.model.odSystem.DeliveryServiceList;
import isep.webapp.terasync.model.odSystem.Product;
import isep.webapp.terasync.model.odSystem.ProductMenu;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.util.List;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Business extends User {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "businessId")
    private int businessId;
    @Column(name = "name")
    private String name;
    @Column(name = "address")
    private String address;
    @Column(name = "phone")
    private String phone;
    @Column(name = "email")
    private String email;
    @Column(name = "username")
    private String username;
    @Column(name = "password")
    private String password;
    @Column(name = "token")
    private String token;
    @Column(name = "registrationDate")
    @CreationTimestamp
    private String registrationDate;
    @Column(name = "emailVerified")
    private boolean emailVerified;
    @Column(name= "approved")
    private boolean approved;
    @Column(name = "pfpImgPath")
    private String pfpImgPath;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = CustomerOrder.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "businessId")
    private List<CustomerOrder> customerOrders;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = DeliveryServiceList.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "businessId")
    private List<DeliveryServiceList> deliveryServiceLists;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = ProductMenu.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "businessId")
    private List<ProductMenu> productMenus;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = Product.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "businessId")
    private List<Product> products;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = RatingList.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "businessId")
    private List<RatingList> ratingLists;

    @Override
    public int getUserId() {
        return businessId;
    }

    @Override
    public boolean getEmailVerified() {
        return emailVerified;
    }
}

File: front/src/app/model/user/business.ts
import {User} from "./user";
import {CustomerOrder} from "../odSystem/customer.order";
import {DeliveryServiceList} from "../odSystem/delivery.service.list";
import {ProductMenu} from "../odSystem/product.menu";
import {Product} from "../odSystem/product";
import {RatingList} from "../rating.list";

export class Business extends User {
  protected override name: string;
  override businessId: number | undefined;
  override address: string;
  override phone: string;
  override approved: boolean | undefined;
  override thumbnail: string | undefined;

  override customerOrders: CustomerOrder[] = [];
  override deliveryServiceLists: DeliveryServiceList[] = [];
  override productMenus: ProductMenu[] = [];
  override products: Product[] = [];
  override ratingLists: RatingList[] = [];

  constructor(name: string, email: string, username: string,
              password: string, address: string, phone: string,
              businessId?: number | undefined,
              registrationDate?: string | undefined, token?: string | undefined,
              emailVerified?: boolean | undefined, approved?: boolean | undefined,
              pfpImgPath?: string | undefined, thumbnail?: string | undefined) {
    super(email, username, password, businessId, registrationDate, token, emailVerified, pfpImgPath, name, thumbnail);
    this.name = name;
    this.businessId = businessId;
    this.address = address;
    this.phone = phone;
    this.approved = approved;
  }

  static override fromJson(jsonBusiness: Business) {
    let business = new Business(jsonBusiness.name, jsonBusiness.email, jsonBusiness.username,
      jsonBusiness.password, jsonBusiness.address, jsonBusiness.phone,
      jsonBusiness.businessId, jsonBusiness.registrationDate, jsonBusiness.token,
      jsonBusiness.emailVerified, jsonBusiness.approved, jsonBusiness.pfpImgPath, jsonBusiness.thumbnail)

    business.customerOrders = CustomerOrder.initializeCustomerOrders(jsonBusiness);
    business.deliveryServiceLists = DeliveryServiceList.initializeDeliveryServiceLists(jsonBusiness);
    business.productMenus = ProductMenu.initializeProductMenus(jsonBusiness);
    business.products = Product.initializeProducts(jsonBusiness);
    business.ratingLists = RatingList.initializeRatingLists(jsonBusiness);

    return business;
  }
}
